src/commands/status.yml

```diff
@@ -167,46 +207,51 @@ steps:
                 echo "Job completed successfully. Alert sent."
               fi
             else
-              #If Failed
-              curl -X POST -H 'Content-type: application/json' \
-                --data "{ \
-                  <<# parameters.channel >>
-                  \"channel\": \"<< parameters.channel >>\", \
-                  <</ parameters.channel >>
-                  \"attachments\": [ \
-                    { \
-                      \"fallback\": \"<< parameters.failure_message >>\", \
-                      \"text\": \"<< parameters.failure_message >> $SLACK_MENTIONS\", \
-                      \"fields\": [ \
-                        <<# parameters.include_project_field >>
-                        { \
-                          \"title\": \"Project\", \
-                          \"value\": \"$CIRCLE_PROJECT_REPONAME\", \
-                          \"short\": true \
-                        }, \
-                        <</ parameters.include_project_field >>
-                        <<# parameters.include_job_number_field >>
-                        { \
-                          \"title\": \"Job Number\", \
-                          \"value\": \"$CIRCLE_BUILD_NUM\", \
-                          \"short\": true \
-                        } \
-                        <</ parameters.include_job_number_field >>
-                      ], \
-                      \"actions\": [ \
-                        <<# parameters.include_visit_job_action >>
-                        { \
-                          \"type\": \"button\", \
-                          \"text\": \"Visit Job\", \
-                          \"url\": \"$CIRCLE_BUILD_URL\" \
-                        } \
-                        <</ parameters.include_visit_job_action >>
-                      ], \
-                      \"color\": \"#ed5c5c\" \
-                    } \
-                  ] \
-                } " << parameters.webhook >>
-              echo "Job failed. Alert sent."
+              if [ << parameters.change_only >> = true ] && [ "success" != "$SLACK_LAST_BUILD_STATUS" ]; then
```

Comment:
Has the outcome been tested for jobs that have been skipped or canceled?

Reply:
Looking into this a bit..

The current slack orb counts `success` as a success, and anything else as a failure.

This means that the following are counted as a failure:

* `canceled`
* `infrastructure_fail`
* `timedout`
* `failed`
* `no_tests`

If we leave with the same logic as it stands it would be the current behavior in the code.  It doesn't seem like the the current orb differentiates between the different failure scenarios.

We _could_ check if it's EXACTLY a failure and in that case it would emit erroneously at some points but that might be counted as acceptable (eg, I'm not sure if `when: on_fail` is fired for something like `timedout`)

Reply:
We are going to close this for now and revisit the topic when we have a more deterministic method of determining changes. We will continue to look into this not only for Slack but this same mechanic is useful in mono-repos.